fluss-common/src/main/java/com/alibaba/fluss/bucketing/BucketingFunction.java

```diff
@@ -20,8 +20,10 @@
 
 import javax.annotation.Nullable;
 
+import java.io.Serializable;
+
 /** An interface to assign a bucket according to the bucket key byte array. */
-public interface BucketingFunction {
+public interface BucketingFunction extends Serializable {
```

Comment:
We can construct `BucketingFunction` and `KeyEncoder` during runtime, to avoid to make so many classes `Serializable`.

Reply:
Done!